### ISSUE TITLE
[FR] Add multi-annotator agreement metrics for human annotation workflows

### ISSUE THREAD

**Issue by debu-sinha** [mlflow/mlflow#22948]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

Add a `compute_annotator_agreement()` function that takes traces with multiple human assessments and computes inter-rater reliability metrics (Cohen's kappa for 2 annotators, Fleiss' kappa for 3+), along with a list of traces where annotators disagree.

### Motivation

> #### What is the use case for this feature?

Before running `scorer.align()`, users need to know if their human annotators agree with each other. If humans disagree on what "correct" means, alignment will learn from noise and produce a bad judge.

> #### Why is this use case valuable to support for MLflow users in general?

Any team collecting human annotations for judge alignment needs this check. It's a standard step in annotation workflows across NLP and ML evaluation.

> #### Why is this use case valuable to support for your project(s) or organization?

Internal judge-building workshops have discovered low inter-rater reliability after running alignment, wasting time and compute. A pre-alignment agreement check would catch this early.

> #### Why is it currently difficult to achieve this use case?

MLflow has no built-in inter-rater reliability computation. Users must manually query assessments from traces, pivot by annotator, and compute kappa using external libraries (scikit-learn or statsmodels). The existing aggregation module (`mlflow/genai/scorers/aggregation.py`) only computes statistical metrics (min, max, mean, median, p90) on scorer outputs, not agreement metrics across annotators.

### Details

Proposed API:

```python
from mlflow.genai.evaluation import compute_annotator_agreement

report = compute_annotator_agreement(
    traces=labeled_traces,
    assessment_name="correctness",
)

report.cohens_kappa          # 0.82 (pairwise, for 2 annotators)
report.fleiss_kappa          # 0.79 (for 3+ annotators)
report.annotator_count       # 3
report.trace_count           # 47
report.disagreement_traces   # [trace_id_12, trace_id_31]
```

Implementation would be ~100 lines using scipy or a lightweight kappa implementation. Main work is the assessment extraction and pivoting logic.

cc @smoorjani - this pairs with judge evaluation metrics (#22947) as a pre-alignment quality gate. Happy to take this on if it looks ready.

**Comment by DASVISHNU** [comment]
Hi @debu-sinha, I would like to work on this feature.
I have handson experience with Python and ML libraries like scikit-learn.

Before starting, I had a quick question:

 Should we rely on external libraries (e.g., sklearn/statsmodels) for kappa computation, or implement lightweight versions internally?

Looking forward to your guidance!

**Comment by serena-ruan** [comment]
I'd keep this issue open for votes

**Comment by debu-sinha** [comment]
Thanks for the interest @DASVISHNU. I plan to take this one on myself since it connects to some related work I have in progress on the evaluation pipeline. Will post a design sketch here before starting the implementation.
